rosbag2_cpp/src/rosbag2_cpp/writers/sequential_writer.cpp

```diff
@@ -203,7 +206,20 @@ void SequentialWriter::write(std::shared_ptr<rosbag2_storage::SerializedBagMessa
   const auto duration = message_timestamp - metadata_.starting_time;
   metadata_.duration = std::max(metadata_.duration, duration);
 
-  storage_->write(converter_ ? converter_->convert(message) : message);
+  // No need to cache if there's only space for 1 message
+  // We'll call write directly in this case.
+  if (chunk_size_ <= 1) {
+    storage_->write(converter_ ? converter_->convert(message) : message);
+    return;
+  }
```

Comment:
I'd prefer this to have the rest of the function written within an else block rather than having a return.

Reply:
should be done.